src/rgw/rgw_op.h

```diff
@@ -1773,7 +1773,8 @@ class RGWGetObjLayout : public RGWOp {
   rgw_raw_obj head_obj;
 
 public:
-  RGWGetObjLayout() {
+  RGWGetObjLayout() = default;
+  ~RGWGetObjLayout() {
```

Comment:
>> how about using std::unique_ptr<RGWRados::Object> target; instead though? then we don't need a constructor or destructor at all

It is good if target is used only inside RGWGetObjLayout::execute() and should be deleted when it leaves the scope. If the scope of target is with in the stack, then why allocating the memory using new operator in RGWGetObjLayout::execute() ?